src/geode/geometry/distance.cpp

```diff
@@ -45,6 +46,22 @@ namespace geode
             Vector< dimension >{ point, nearest_p }.length(), nearest_p );
     }
 
+    std::tuple< double, Point2D > point_segment_signed_distance(
+        const Point2D& point, const Segment2D& segment )
+    {
+        Point2D nearest_point;
+        double distance;
+        std::tie( distance, nearest_point ) =
+            point_segment_distance< 2 >( point, segment );
+        const Vector2D proj2point{ nearest_point, point };
+        // segment 2D facet normals point towards inside
+        const auto signed_distance =
+            dot_perpendicular( proj2point, segment.direction() ) <= 0
```

Comment:
should we use `dot_perpendicular` or `triangle_signed_area` using query and segment extremities? For info: `dot_perpendicular` is used in `triangle_signed_area`: 
```        return dot_perpendicular(
                   { triangle.vertices()[0], triangle.vertices()[1] },
                   { triangle.vertices()[0], triangle.vertices()[2] } )
               / 2.;
```
It could prevent being unhomogoneous.

Reply:
True

Reply:
I would use query point as Triangle vertex 0 and segment extremities being the two other vertices in the same order